apps/meteor/app/livechat/server/api/v1/contact.js

```diff
@@ -51,19 +51,23 @@ API.v1.addRoute(
 				check(this.queryParams, {
 					email: Match.Maybe(String),
 					phone: Match.Maybe(String),
+					custom: Match.Maybe(String),
 				});
+				const { email, phone, custom } = this.queryParams;
 
-				const { email, phone } = this.queryParams;
-
-				if (!email && !phone) {
+				if (!email && !phone && !custom) {
 					throw new Meteor.Error('error-invalid-params');
 				}
-
+				const allowedCF = LivechatCustomField.find({ scope: 'visitor' }, { fields: { _id: 1 } }, { searchability: true }).map(
+					({ _id }) => _id,
+				);
+				const customObj = Object.fromEntries(Array.from(new URLSearchParams(custom)).filter(([k]) => allowedCF.includes(k)));
```

Comment:
If you do this the other way, you can avoid having to query _all_ custom fields for visitor 👀 
Like:
1. Parse received custom fields
2. Validate they're not something nefarious (string only)
3. Find Custom Fields that match the received ones
4. Use the `found` list to do the queries